code/scorpio/troll.dm

```diff
@@ -0,0 +1,120 @@
+// troll.dm
+// Copyright 2021 Patrick Meade.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//------------------------------------------------------------------------------
+
+GLOBAL_LIST_EMPTY(gamer_words)
+
+/proc/add_gamer_words(config_value)
+	GLOB.gamer_words += splittext(config_value, ",")
+
+/proc/ban_and_kick_troll(context, message, word, mob_name, client/ban_me)
+	ban_troll(context, message, word, mob_name, ban_me)
+	kick_troll(ban_me)
+
+/proc/ban_troll(context, message, word, mob_name, client/ban_me)
+	// we can't ban if we don't have a database
+	if(!SSdbcore.IsConnected())
+		log_and_message_admins("Database connection failure when attempting to ban troll [ban_me].")
+		return FALSE
+
+	// build up all the things we need to record the ban
+	var/serverip = "[world.internet_address]:[world.port]"
+	var/bantype_str = "PERMABAN"
+	var/reason = "[ban_me] / ([mob_name]) used gamer word '[word]' in [context] message [message]"
+	var/job = ""
+	var/duration = -1 // PERMABAN
```

Comment:
Instantly banning isnt a good idea here. Give a "3 strikes and youre out" policy